src/platforms/common/stm32/dfu_f4.c

```diff
@@ -41,7 +41,29 @@ static uint32_t sector_addr[] = {
 	0,
 };
 
-static uint16_t sector_erase_time[] = {500, 500, 500, 500, 1100, 2600, 2600, 2600, 2600, 2600, 2600, 2600, 2600};
+/* Sector erase times in milliseconds, typ, for x32 parallelism at 2.7-3.6v */
+typedef enum erase_times_f4 {
+	ERASE_TIME_16KB = 250,   /*  500 * 0.5 */
+	ERASE_TIME_64KB = 550,   /* 1100 * 0.5 */
+	ERASE_TIME_128KB = 1000, /* 2000 * 0.5 */
```

Comment:
This decreases the timing from 2600 to 2000 - what's the basis for this?

Reply:
* DS10314 STM32F411xC/xE Rev 7 pg92, Table 45. Flash memory programming
* DS8626 STM32F405xx/407xx Rev 9 pg108, Table 40. Flash memory programming
* DS9405 STM32F427xx/429xx Rev 10 pg129, Table 48. Flash memory programming

Should I hook the datasheets into the dfu_f4.c code comments?
2600ms is max t_ERASE128KB at PSIZE=x16, we're using x32, so 2000ms is more appropriate.

Reply:
Yes, please - that's exactly the sort of information we've love to have in the source for this exact reason and question 🙂 - we'll mark this as resolved once comments for each one linking the DS number, revision, page number and table number are added much like in your reply are in the source for future reference.

Reply:
Listed datasheet numbers and ST automatic URLs after top block comment. Also listed platforms corresponding to them. That is f4discovery, all blackpill-f4, and stlinkv3. Others seem to not use it: 96b_carbon, hydrabus. These platforms may have a different flash IP: f072_if, f3_if.